src/Fcf/Data/Symbol.hs

```diff
@@ -0,0 +1,115 @@
+{-# LANGUAGE DataKinds              #-}
+{-# LANGUAGE PolyKinds              #-}
+{-# LANGUAGE TypeFamilies           #-}
+{-# LANGUAGE TypeInType             #-}
+{-# LANGUAGE TypeOperators          #-}
+{-# LANGUAGE UndecidableInstances   #-}
+{-# OPTIONS_GHC -Wall                       #-}
+{-# OPTIONS_GHC -Werror=incomplete-patterns #-}
+
+{-|
+Module      : Fcf.Data.Symbol
+Description : Type level symbols
+
+= Fcf.Data.Symbol
+
+Type-level symbols and functions for them.
+
+Note that the operators from this module conflict with "GHC.TypeLits".
+
+-}
+
+--------------------------------------------------------------------------------
+
+module Fcf.Data.Symbol
+    ( -- * Reexported type
+      -- | From "GHC.TypeLits".
+
+      Symbol
+
+     -- * Comparison functions
+
+    , SymbolOrd
+    , type (<=)
+    , type (>=)
+    , type (<)
+    , type (>)
+    , type (==)
+    )
+  where
+
+--------------------------------------------------------------------------------
+        
+import           GHC.TypeLits (Symbol)
+import qualified GHC.TypeLits as TL
+
+import           Fcf.Core (Eval, Exp)
+import           Fcf.Data.Bool (type (||))
+import           Fcf.Utils (TyEq)
+
+--------------------------------------------------------------------------------
+
+
+-- | SymbolOrd - compare two symbols and give type-level Ordering 
+-- ( $ 'LT $, $ 'EQ $ or $ 'GT $ ).
+--
+-- === __Example__
+-- 
+-- >>> :kind! Eval (SymbolOrd "a" "b")
+-- Eval (SymbolOrd "a" "b") :: Ordering
+-- = 'LT
+data SymbolOrd :: Symbol -> Symbol -> Exp Ordering
```

Comment:
Let's create a general module `Fcf.Class.Ord` for this instead.

Reply:
Comparing this to Functor and fmap, do you mean Ord and compare? Which would mean that I'll write compare function and separate instances for it.

Reply:
I followed this plan and made changes